shared/macros/10-bash.jinja

```diff
@@ -1334,7 +1334,9 @@ printf -v formatted_output "{{{ format }}}" "$stripped_key" "{{{ value }}}"
 # We search for the key string followed by a word boundary (matched by \>),
 # so if we search for 'setting', 'setting2' won't match.
 if LC_ALL=C grep -q -m 1 -i -e "{{{ key }}}\\>" "{{{ config_file }}}"; then
-    "${sed_command[@]}" "s/{{{ key }}}\\>.*/$formatted_output/gi" "{{{ config_file }}}"
+    escaped_key=$(sed -e 's|/|\\/|g' <<< "{{{ key }}}")
+    escaped_formatted_output=$(sed -e 's|/|\\/|g' <<< "$formatted_output")
```

Comment:
There is a lot of unit tests for the `bash_replace_or_append` in `tests/unit/bash/test_bash_replace_or_append.bats.jinja`. Could you add there a unit test that covers this situation (in which key contain slashes)?

Reply:
I searched the project and just found one rule using slashes on the key, i.e. `rsyslog_remote_loghost`:
```
{{{ bash_replace_or_append('/etc/rsyslog.conf', '^\*\.\*', "@@$rsyslog_remote_loghost_address", '%s %s') }}}
```
However backward slashes don't cause `sed` to fail.
I updated the change to only escape the value and not the key. Added a test to cover this scenario (value contains forward slashes).